src/CodeStyle/Core/Analyzers/AbstractFormattingAnalyzer.cs

```diff
@@ -1,44 +1,51 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
 using System.IO;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.VisualStudio.CodingConventions;
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
-    internal abstract class AbstractFormattingAnalyzerImpl
+    internal abstract class AbstractFormattingAnalyzer
+        : AbstractCodeStyleDiagnosticAnalyzer
     {
-        private readonly DiagnosticDescriptor _descriptor;
-
-        protected AbstractFormattingAnalyzerImpl(DiagnosticDescriptor descriptor)
+        protected AbstractFormattingAnalyzer()
+            : base(
+                IDEDiagnosticIds.FormattingDiagnosticId,
+                new LocalizableResourceString(nameof(CodeStyleResources.Fix_formatting), CodeStyleResources.ResourceManager, typeof(CodeStyleResources)),
+                new LocalizableResourceString(nameof(CodeStyleResources.Fix_formatting), CodeStyleResources.ResourceManager, typeof(CodeStyleResources)))
         {
-            _descriptor = descriptor;
         }
 
-        internal void InitializeWorker(AnalysisContext context)
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+        protected abstract ISyntaxFormattingService SyntaxFormattingService { get; }
+
+        protected override void InitializeWorker(AnalysisContext context)
         {
-            var workspace = new AdhocWorkspace();
             var codingConventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
-
-            context.RegisterSyntaxTreeAction(c => AnalyzeSyntaxTree(c, workspace, codingConventionsManager));
+            context.RegisterSyntaxTreeAction(c => AnalyzeSyntaxTree(c, codingConventionsManager));
         }
 
         protected abstract OptionSet ApplyFormattingOptions(OptionSet optionSet, ICodingConventionContext codingConventionContext);
 
-        private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context, Workspace workspace, ICodingConventionsManager codingConventionsManager)
+        private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context, ICodingConventionsManager codingConventionsManager)
         {
             var tree = context.Tree;
             var cancellationToken = context.CancellationToken;
 
-            var options = workspace.Options;
+            OptionSet options = CompilerAnalyzerConfigOptions.Empty;
             if (File.Exists(tree.FilePath))
             {
                 var codingConventionContext = codingConventionsManager.GetConventionContextAsync(tree.FilePath, cancellationToken).GetAwaiter().GetResult();
```

Comment:
so.. editorconfig file is not part of snapshot? that violates roslyn's snapshot model.

Reply:
hmmm... or I might be missing something. there is ICodeConventionSnapshot so, there is something that tie these 2 snapshots?

Reply:
Correct. Editorconfig will not be part of the Roslyn snapshot until the work from project system is complete. This has been the case since the introduction of editorconfig support in Visual Studio.